### ISSUE TITLE
High memory usage

### ISSUE THREAD

**Issue by sfrieson** [zed-industries/community#1186]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I was using Zed yesterday and out of habit switched back to using VS Code.

Today, I got an OS alert that the computer was out of memory.

![Screen Shot 2023-03-17 at 11 47 20 AM](https://user-images.githubusercontent.com/15187179/225955445-09cc66cc-99da-4c90-b6cc-24796cb56da1.png)

I did as instructed and closed a bunch of applications, but going into the Activity Monitor I noticed that Zed's memory usage was extraordinary at 108GB.

<img width="663" alt="image" src="https://user-images.githubusercontent.com/15187179/225955753-5f62c152-c0ae-43e4-850f-8e9998eaf210.png">

### Environment

Zed: v0.77.3 (stable)
OS: macOS 12.6.0
Memory: 32 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by mikayla-maki** [comment]
Thank you for the bug report :)

By chance, did you have Zed open in your home directory, or a large directory in general?

**Comment by JosephTLyons** [comment]
Hey @sfrieson, do you recall any of the details of what you were doing in Zed?  I see a JS project in the background - any other projects open in Zed?  What languages were being used?  This is a pretty bad issue on our end and I'm just looking for any extra details.

**Comment by sfrieson** [comment]
Hey Mikayla and Joseph,

It's a little hard to say what I was doing considering I was in not actively using Zed but generally this is what i was doing over the past day.

Both Zed and VS Code were opened to the same monorepo. The monorepo predominantly contains TypeScript and Rust run with Vite. I normally just run the front-end code, but I had to make some full stack (Node.js, TypeScript) changes, so I started the full app this morning. As you might expect there are rebuilds on each change. This open directory has 150k+ files.

I know for sure one of the open files in Zed was the same file I was updated in VSCode.

Hope some of this helps, and I'm excited to keep trying out Zed!

**Comment by uzaxirr** [comment]
I encountered the same thing. 

<img width="461" alt="Screenshot 2023-03-20 at 11 47 33 AM" src="https://user-images.githubusercontent.com/72073401/226285399-e099b111-205a-4820-98fa-8c44d542743a.png">

I was working with python with a monorepo which is a backend server in Django. And Zed wasn't opened in my home directory.

**Comment by maxbrunsfeld** [comment]
Hi everyone, sorry for the terrible experience.

One reason this might happen is that your project contains symlinks that points to large directories outside of the project. For anyone who is experiencing this problem in a certain project, would you be willing to report what symlinks exist in the project?

This command (run from the root of your project) will list all of the symlinks in the project and where they point:

```sh
while read symlink; do echo $symlink "->" $(readlink "$symlink"); done < <(find . -type l)
```

**Comment by uzaxirr-fam** [comment]
> Hi everyone, sorry for the terrible experience.
> 
> One reason this might happen is that your project contains symlinks that points to large directories outside of the project. For anyone who is experiencing this problem in a certain project, would you be willing to report what symlinks exist in the project?
> 
> This command (run from the root of your project) will list all of the symlinks in the project and where they point:
> 
> ```shell
> while read symlink; do echo $symlink "->" $(readlink "$symlink"); done < <(find . -type l)
> ```

Sure here it is
https://app.warp.dev/block/jczzqLaazfhZuBr17OA4ZK

**Comment by sfrieson** [comment]
Yes, we have many symlinks in the monorepo, about 9700. Many of them seem to come from the `pnpm` manages packages.

**Comment by stevenshih1997** [comment]
Is there a fix for this when a project has a large number of symlinks? I am also running into this issue and prevents me from using zed

**Comment by maxbrunsfeld** [comment]
We’re working on this issue right now. I’ll say something here, when I know which version will contain the fix.

**Comment by maxbrunsfeld** [comment]
I think this will be fixed in Zed 0.93. That will ship *next* Wednesday.

**Comment by JosephTLyons** [comment]
0.93 is now in stable. I'm going to close this issue - please reopen and @ me if the issue persists.
